src/pytest_plugins/consume/consume.py

```diff
@@ -68,6 +70,51 @@ def download_and_extract(url: str, base_directory: Path) -> Path:
     return extract_to / "fixtures"
 
 
+@dataclass
+class SimLimitBehavior:
+    """Stores options derived from the `--sim.limit` argument."""
+
+    collectonly: bool = False
+    pattern: str = ".*"
+
+
+def process_sim_limit(pattern: str) -> SimLimitBehavior:
+    """
+    Process the `--sim.limit` argument to determine test filtering behavior.
+
+    The function handles three cases:
+    1. If the value is exactly "collectonly", enable collection mode without
+        filtering.
+    2. If the value starts with "collectonly:", enable collection mode and use the
+        rest as a regex pattern.
+    3. If the (possibly transformed) value starts with "id:", treat the rest as a
+        literal test ID and escape special regex characters.
+
+    Args:
+        pattern: The raw string passed to the --sim.limit option.
+
+    Returns:
+        A SimLimitBehavior object with appropriate settings.
+
+    """
+    if pattern == "collectonly":
+        return SimLimitBehavior(collectonly=True)
+
+    sim_limit = SimLimitBehavior(pattern=pattern)
+
+    collect_only_prefix = "collectonly:"
```

Comment:
This would be a lot easier if there was a way to pass custom environment variables to the simulator from the hive executable.

Reply:
I'm a bit wary of adding generic per-simulator args/config (even thought that's what we're doing here :laughing:). A  simple, consistent UX is quite nice for people that don't use it regularly. Having said that, there will definitely be other use cases for this.

I do think we should discuss/explore whether we can add other standard flags to hive (like --collect-only, for all simulators).